### PR TITLE
Implementing full psr/http-message and php 7.4

### PR DESCRIPTION
**Pull request by byjg**

### CODE REVIEW

**Comment by GeoDpto** [comment]
@byjg Hi!

Are there any updates for this pull request?

File: README.md [hunk 1]
```diff
@@ -22,13 +21,14 @@ This class implements:
...
 
 The key elements are:
+
 * URI - Will define the URI with parameters, path, host, schema, etc
 * Request - Will set the request headers and method;
-* Response - Will receive the response header, body and status code. 
+* Response - Will receive the response header, body and status code.
 
-More information about the PSR-7 here: https://www.php-fig.org/psr/psr-7/
+More information about the PSR-7 here: <https://www.php-fig.org/psr/psr-7/>
```
Comment (vitormattos) [inline README.md:29]: Only a suggestion
```suggestion
More information about the PSR-7 [here](https://www.php-fig.org/psr/psr-7/).
```

**Comment by casingh1990** [comment]
Hi Team, 

This is causing an issue for us in a Laravel 10.x project. 

Laravel 10.x requires Guzzle's `guzzlehttp/psr7` version 2.6.2. 
Dependency tree is: Laravel -> monolog/monolog -> guzzlehttp/psr7 -> psr/http-message (1.1 or 2.0)

If this PR is approved, then it appears we'll all be at psr/http-message 2.0 and everything _should_ work. 

Any ETA on this?

File: tests/Psr7/NullStreamTest.php [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+    public function testIsWritable()
+    {
+        $this->assertTrue($this->stream->isWritable());
+    }
+
+    public function testOverwrite()
+    {
+        $this->stream->rewind();
+        $result = $this->stream->write("test");
```
Comment (vitormattos) [inline tests/Psr7/NullStreamTest.php:95]: You mixed single and double quotes at this file here and in other places

```suggestion
        $result = $this->stream->write('test');
```

**Comment by vitormattos** [comment]
Worked fine to me.

**Comment by byjg** [comment]
This PR is breaking change for now. 

Renamed namespace `ByJG\Util` to `ByJG\WebRequest` 

I am updating all byjg dependent classes, **please, upgrade your versions to `^5.0` or `5.0.x-dev`**, **_all of byjg packages_**

**Pull request merged**